apps/examples/testcase/ta_tc/mqtt/utc/utc_mqtt_main.c

```diff
@@ -0,0 +1,479 @@
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         utc_mqtt_unsubscribe_p
+ * @brief            stop subscribing from a broker
+ * @scenario         call mqtt_unsubscribe
+ * @apicovered       mqtt_unsubscribe
+ * @precondition     none
+ * @postcondition    none
+ */
+void utc_mqtt_unsubscribe_p(void)
+{
+	int res = mqtt_unsubscribe(g_mqtt_sub_handle, UTC_MQTT_TOPIC);
+	TC_ASSERT_EQ("mqtt_unsbscribe", res, 0);
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         utc_mqtt_disconnect_n
+ * @brief            disconnect to broker
+ * @scenario         disconnect to broker using an invalid handle
+ * @apicovered       mqtt_disconnect
+ * @precondition     none
+ * @postcondition    none
+ */
+void utc_mqtt_disconnect_n(void)
+{
+	int res = mqtt_disconnect(NULL);
+	TC_ASSERT_LT("mqtt_disconnect", res, 0);
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         utc_mqtt_disconnect_p
+ * @brief            disconnect to broker
+ * @scenario         disconnect to broker using a valid handle
+ * @apicovered       db_init
+ * @precondition     none
+ * @postcondition    none
+ */
+void utc_mqtt_disconnect_p(void)
+{
+	int res = mqtt_disconnect(g_mqtt_sub_handle);
+	TC_ASSERT_EQ("mqtt_disconnect", res, 0);
+
+	UTC_MQTT_WAIT_SIGNAL;
+	
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         utc_mqtt_deinit_client_n
+ * @brief            free all resources used to mqtt stack
+ * @scenario         call mqtt_deinit with an invalid input
+ * @apicovered       mqtt_deinit_client
+ * @precondition     none
+ * @postcondition    none
+ */
+void utc_mqtt_deinit_client_n(void)
+{
+	int res = mqtt_deinit_client(NULL);
+	TC_ASSERT_LT("mqtt_deinit_client", res, 0);
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+ * @testcase         utc_mqtt_deinit_client_p
+ * @brief            free all resources used to mqtt stack
+ * @scenario         call mqtt_deinit with an invalid input
+ * @apicovered       mqtt_deinit_client
+ * @precondition     none
+ * @postcondition    none
+ */
+void utc_mqtt_deinit_client_p(void)
+{
+	int res = mqtt_deinit_client(g_mqtt_pub_handle);
+	if (res < 0) {
+		UTC_MQTT_LOGE;
+	}
+	
+	res = mqtt_deinit_client(g_mqtt_sub_handle);
+	TC_ASSERT_EQ("mqtt_deinit_client", res, 0);
+	TC_SUCCESS_RESULT();
+}
+
+
+int mqtt_utc(int argc, FAR char *argv[])
+{
+	int res = _utc_mqtt_init();
+	if (res < 0) {
+		UTC_MQTT_LOGE;
+		return -1;
+	}
+	
+	utc_mqtt_init_client_n();
+	utc_mqtt_init_client_p();
+	utc_mqtt_connect_n();
+	utc_mqtt_connect_p();
+	utc_mqtt_subscribe_n();
+	utc_mqtt_subscribe_p();
+	utc_mqtt_publish_n();
+	utc_mqtt_publish_p();
+	utc_mqtt_unsubscribe_n();
+	utc_mqtt_unsubscribe_p();
+	utc_mqtt_disconnect_n();
+	utc_mqtt_disconnect_p();
+	utc_mqtt_deinit_client_n();
+	utc_mqtt_deinit_client_p();
+
+	_utc_mqtt_deinit();
+
+	return 0;
+}
+
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int utc_mqtt_main(int argc, char *argv[])
+#endif
+{
+
+#ifdef CONFIG_TASH
+	tash_cmd_install("mqtt_utc", mqtt_utc, TASH_EXECMD_SYNC);
```

Comment:
This makes tester should type in a command to lauch this. I think we don't need this.

Reply:
I think a mqtt test case needs it

Reply:
1. When user executes testcase example, all of testcase should be run by itself. MQTT doesn't have any limitation to do it. Why do we run it as a TASH command seperately?
2. If you want to run it as a TASH command due to some reason, we don't need to make a MQTT task definitely.
 Just register it at tc_main function.